core/templates/dev/head/domain/state_card/StateCardObjectFactory.ts

```diff
@@ -17,199 +17,231 @@
-    var StateCard = function(
-        stateName, contentHtml, interactionHtml, interaction,
-        inputResponsePairs, recordedVoiceovers, contentId) {
-      this._stateName = stateName;
-      this._contentHtml = contentHtml;
-      this._interactionHtml = interactionHtml;
-      this._inputResponsePairs = inputResponsePairs;
-      this._interaction = interaction;
-      this._recordedVoiceovers = recordedVoiceovers;
-      this._contentId = contentId;
-      this._completed = false;
-    };
-
-    StateCard.prototype.getStateName = function() {
-      return this._stateName;
-    };
-
-    StateCard.prototype.getInteraction = function() {
-      return this._interaction;
-    };
-
-    StateCard.prototype.getVoiceovers = function() {
-      var recordedVoiceovers = this._recordedVoiceovers;
-      var contentId = this._contentId;
-      if (recordedVoiceovers) {
-        return recordedVoiceovers.getBindableVoiceovers(contentId);
-      }
-      return [];
-    };
-
-    StateCard.prototype.getRecordedVoiceovers = function() {
-      return this._recordedVoiceovers;
-    };
-
-    StateCard.prototype.isContentAudioTranslationAvailable = function() {
-      return Object.keys(
-        this.getVoiceovers()).length > 0 ||
-        AudioTranslationLanguageService.isAutogeneratedAudioAllowed();
-    },
-
-    StateCard.prototype.getInteractionId = function() {
-      if (this.getInteraction()) {
-        return this.getInteraction().id;
-      }
-      return null;
-    };
-
-    StateCard.prototype.isTerminal = function() {
-      var interactionId = this.getInteractionId();
-      return (
-        interactionId && INTERACTION_SPECS[interactionId].is_terminal);
-    };
-
-    StateCard.prototype.getHints = function() {
-      return this.getInteraction().hints;
-    };
-
-    StateCard.prototype.getSolution = function() {
-      return this.getInteraction().solution;
-    };
-
-    StateCard.prototype.doesInteractionSupportHints = function() {
-      var interactionId = this.getInteractionId();
-      return (
-        !INTERACTION_SPECS[interactionId].is_terminal &&
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { AppConstants } from 'app.constants';
+import { AudioTranslationLanguageService } from
+  'pages/exploration-player-page/services/audio-translation-language.service';
+import { Interaction } from 'domain/exploration/InteractionObjectFactory';
+import { RecordedVoiceovers } from
+  'domain/exploration/RecordedVoiceoversObjectFactory';
+
+const INTERACTION_SPECS = require('interactions/interaction_specs.json');
+
+export interface IInputResponsePair {
+  learnerInput: string,
+  oppiaResponse: string,
+  isHint: boolean
+}
+
+export class StateCard {
+  _stateName: string;
+  _contentHtml: string;
+  _interactionHtml: string;
+  _interaction: Interaction;
+  // TODO(#7165): Replace 'any' with the exact type.
```

Comment:
Is this TODO needed here? I don't see `any` here.

Reply:
Ah sorry. I added the type later. Thanks! fixed.